accepted/future-releases/constructor-tearoffs/faq.md

```diff
@@ -0,0 +1,248 @@
+
+Example:
+
+```dart
+class C<T> {
+  const C.new();
+  C.named() : this.new();
+  /// Calls [C.new].
+  factory C.otherNamed() = C<T>.new;  
+}
+class D {
+  D() : super.new();
+}
+void main() {
+  var cs = [C<int>.new(), const C<int>.new(), new C<int>.new()];
+  var tearoff = C.new; // New!
+  var explicitlyTypedTearoff = C<int>.new; // New!
+}
+```
+
+Everywhere except the tear-offs, you can remove the `.new`  and it means the same thing. The tearoff is the only place which *requires* the `.new`.
+
+**Q:** Should I use `.new` or not. What does the style guide say?
+
+**A:** The style guide says nothing yet. The general rule is to not write something which isn't necessary, so don't write `new` unless you need to. For now, only use it for tear-offs and possibly DartDoc (it will be more Dart-like to write `[C.new]` than the current DartDoc-only `[C.C]` to refer to the unnamed constructor). _Don't ever write `new Foo.new()`._
+
+**Q:** Why introduce this everywhere when it's only supposed to be used for tear-offs. Couldn't it just work for tear-offs?
+
+**A:** For consistency *and* because it's expected to be useful for other things too, like generic constructors where we'll also need a way to add type arguments to both the class and the constructor. We'd rather introduce a full feature once than a partial feature now and then having to add another part to it later.
+
+**Q:** Can I declare both `C` and `C.new` in the same class?
+
+**A:** No. The *name* of the constructor is still `C`, the `C.new` is just another syntax for declaring a constructor named `C`, and you still can't declare two constructors with the same name.
+
+**Q:** Will `dart:mirrors` be able to see the `.new` on a constructor declaration?
+
+**A:** Most likely not. There are no plans to change the `dart:mirrors`, and it's just different syntax for the same declaration. The constructor name is still going to be just the class name, and that's what `dart:mirrors` expose.
+
+## Function value instantiation
+
+Until now you've could instantiate *tear-offs* of function declarations or instance methods. You could write:
+
+```dart
+T id<T>(T value) => value;
+int Function(int) intId = id; // implicitly instantiated with <int>.
+```
+
+but you couldn't instantiate function *values* with a type argument:
+
+```dart
+T Function<T>(T) id = <T>(T value) => value;
+int Function(int) intId = id; // INVALID
+```
+
+There were reasons for this, mainly worries about implementations not being able to be efficient. The implementors have told us that it's not a problem, so we remove that restriction and allow you to instantiate any function-typed expression. The `INVALID` above becomes valid and well-typed.
+
+This also applies to *callable objects* (objects which has an interface type with a `call` method), which we treat like function values in most places.
```

Comment:
objects which has an interface type

=>

objects that have an interface type